### PR TITLE
Restructure t_res

### PR DESCRIPTION
**Pull request by rwood-97**
This PR:
- Restructures the T-Res repo so source files are in `t_res` dir
- Modules can now be imported as `import t_res.geoparser.xxx` - this is updated across all files
- Updates docs to reflect new structure
- Update example notebooks to reflect new structure
- Add `-p`/`--path` flag to experiment modules so that resources dir can be specified

### CODE REVIEW

**Comment by thobson88** [comment]
This is almost ready to merge, but when I do a fresh checkout and install with poetry I get:
```
> poetry install
Installing dependencies from lock file

Package operations: 113 installs, 0 updates, 0 removals

  • Installing scipy (1.6.1): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel

  Error: 'numpy' must be installed before running the build.
```
One solution to this is to require Python 3.9 by editing `pyproject.toml` from this:
```
python = ">=3.9, <4.0"
```
to this:
```
python = ">=3.9, <3.10"
```
@rwood-97 do you know if there's a better solution?

**Comment by rwood-97** [comment]
Can you try changing the scipy line in pyproject.toml to:

scipy = "^1.11.0"

**Comment by rwood-97** [comment]
Definitely we are capping torch at a version which only supports up to python 3.10, I wonder if we should update it or cap the python version?

**Comment by rwood-97** [comment]
I pushed a change, I thiink it should fix it

**Comment by thobson88** [comment]
> I pushed a change, I thiink it should fix it

Brilliant - thanks.

File: tests/test_disambiguation.py [hunk 1]
```diff
@@ -143,9 +146,9 @@ def test_train(tmp_path):
     )
 
     # assert expected performance on test set
-    assert mylinker.rel_params["ed_model"].best_performance["f1"] == 0.6288416075650118
+    assert mylinker.rel_params["ed_model"].best_performance["f1"] == 0.8571428571428571
```
Comment (rwood-97) [inline tests/test_disambiguation.py]: https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-approx

use 0.1 + 0.2 == pytest.approx(0.3)

**Pull request merged**